autrite/extract_rule.py

```diff
@@ -0,0 +1,141 @@
+import rule
+
+
+class ExtractQueryRule(rule.Rule):
```

Comment:
Just a minor thought:
How do you handle if one query has two inclusion constraint fields?
for example, q = select * from R where a = ? and b = ?
It seems you return [q]
The behavior looks fine, just want to make sure I understand it correctly

Reply:
if table R both contains both a and b with inclusion constraint, my algorithm will detect this query has inclusion constraint when scan to the part "where a = ?", and return [q] from that. 

I think this works fine because we only want to know if a query contains inclusion constraint or not, regardless how many inclusion constraints.